torch/distributions/constraints.py

```diff
@@ -44,13 +59,22 @@ def check(self, x):
 
 
 def is_dependent(constraint):
```

Comment:
What are applications in mind for `is_dependent` and `dependent_property`?

Reply:
They're useful in generic programming as a kind of NotImplemented or NeedMoreInfo class. Here's an example:
```py
def initialize_params(random_variables):
    params = {}
    for var_name, dist in random_variables.items():
        assert isinstance(dist, Distribution)
        for param_name, constraint in dist.params.items():
            name = '{}.{}'.format(var_name, param_name)
            if is_dependent(constraint):                                   # <------ Example
                raise ValueError('cannot initialize constraints for dependent param {}'.format(name))
            params[name] = transform(constraint).to_constrained(torch.normal(1))
    return params
```

Reply:
That makes sense. I guess it's necessary because `_Dependent` isn't part of the public namespace.

That example is interesting. Because it seems like you _can_ register a transform that's a function of the dependent type's parameters.